libs/threading_base/include/hpx/threading_base/scheduler_base.hpp

```diff
@@ -251,8 +251,9 @@ namespace hpx { namespace threads { namespace policies {
             switch (stacksize)
             {
             case thread_stacksize_small:
+                HPX_FALLTHROUGH;
+            case thread_stacksize_current:
```

Comment:
Did you have a case where `stacksize` still was `thread_stacksize_current` at this point? I would rather put an assert in the first if after `stacksize` has been updated as this shouldn't happen. A created HPX thread should always have one of the actual sizes (small, medium, ...), and a non-HPX thread should return the default which is the small stacksize at the moment.

Reply:
Yes, that's what I was seeing. I did add an assert inside the `if()`, which fired for me. I tried to investigate but was not able to figure out why this happens.

Reply:
Hrm, I'm clearly wrong in that case. I think what we should do is reset a `thread_stacksize_current` to the a more concrete size when creating the thread. This is most likely a bug from my earlier changes. We can go with this and I'll try to fix it afterwards.

Reply:
Yes, I don't like this 'fix' either. See #4648 for a separate ticket.